persistence/db.go

```diff
@@ -26,50 +26,108 @@
 package persistence
 
 import (
-	"fmt"
-
 	"clouditor.io/clouditor/api/auth"
 	"clouditor.io/clouditor/api/orchestrator"
+	"fmt"
 	"github.com/sirupsen/logrus"
 	"gorm.io/driver/postgres"
 	"gorm.io/driver/sqlite"
 	"gorm.io/gorm"
 )
 
 var log *logrus.Entry
-var db *gorm.DB
+
+// TODO(lebogg): Lack for better term
+type GormX struct {
+	db *gorm.DB
+}
 
 func init() {
 	log = logrus.WithField("component", "db")
 }
 
-func InitDB(inMemory bool, host string, port int16) (err error) {
+func (g *GormX) Init(inMemory bool, host string, port int16) (err error) {
```

Comment:
Can we change this to a `NewXXX` style builder? Similar to what we have in the services? 

`var db = NewGORM(WithInMemory())`

`var db = NewGORM(WithPostgres("localhost", "user", "password))`

In both cases ´db` should then be of the interface type `Database`